src/library/Uno.Themes.WinUI.Markup/Theme/TextBox.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+using Uno.Extensions.Markup.Internals;
+using Windows.UI.Text;
+
+namespace Uno.Themes.Markup
+{
+	public static partial class Theme
+	{
+		public static class TextBox
+		{
+			public static class Resources
+			{
+				public static class DeleteButton
+				{
+					public static class Foreground
+					{
+						[ResourceKeyDefinition(typeof(Brush), "TextBoxDeleteButtonForeground")]
+						public static ResourceValue<Brush> Default => new("TextBoxDeleteButtonForeground", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "TextBoxDeleteButtonForegroundPointerOver")]
+						public static ResourceValue<Brush> PointerOver => new("TextBoxDeleteButtonForegroundPointerOver", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "TextBoxDeleteButtonForegroundPressed")]
+						public static ResourceValue<Brush> Pressed => new("TextBoxDeleteButtonForegroundPressed", true);
+
+						[ResourceKeyDefinition(typeof(Brush), "TextBoxDeleteButtonForegroundDisabled")]
+						public static ResourceValue<Brush> Disabled => new("TextBoxDeleteButtonForegroundDisabled", true);
+					}
+				}
```

Comment:
Since Foreground is the only property here for DeleteButton, let's just rename this class to `DeleteButtonForeground` and then have the `Default`/`PointerOver`/etc. properties directly on that class instead of nesting a second `Foreground` class